query-service-impl/src/test/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverterTest.java

```diff
@@ -685,35 +687,31 @@ public void testQueryWithBytesColumnWithEmptyId() {
   }
 
   @Test
-  public void testQueryWithBytesColumnWithLikeFilter() {
+  public void testQueryWithBytesColumnInFilter() {
     Builder builder = QueryRequest.newBuilder();
-    ColumnIdentifier spanId = ColumnIdentifier.newBuilder().setColumnName("Span.id").build();
-    builder.addSelection(Expression.newBuilder().setColumnIdentifier(spanId).build());
-
-    ColumnIdentifier parentSpanId = ColumnIdentifier.newBuilder()
-        .setColumnName("Span.attributes.parent_span_id").build();
-    Filter likeFilter =
-        Filter.newBuilder()
-            .setOperator(Operator.LIKE)
-            .setLhs(Expression.newBuilder().setColumnIdentifier(parentSpanId).build())
-            .setRhs(
-                Expression.newBuilder()
-                    .setLiteral(
-                        LiteralConstant.newBuilder()
-                            .setValue(Value.newBuilder().setString("null").build()))
-                    .build())
-            .build();
+    ColumnIdentifier durationColumn = ColumnIdentifier.newBuilder()
+        .setColumnName("Span.metrics.duration_millis").build();
+    builder.addSelection(Expression.newBuilder().setColumnIdentifier(durationColumn));
+
+    ColumnIdentifier.Builder spanId = ColumnIdentifier.newBuilder().setColumnName("Span.id");
+    // Though span id is bytes in Pinot, top layers send the value as hex string.
+    Value.Builder value = Value.newBuilder().setValueType(ValueType.STRING_ARRAY)
+        .addAllStringArray(List.of("042e5523ff6b2506"));
+    builder.setFilter(Filter.newBuilder()
+        .setLhs(Expression.newBuilder().setColumnIdentifier(spanId))
+        .setOperator(Operator.IN)
+        .setRhs(Expression.newBuilder().setLiteral(LiteralConstant.newBuilder().setValue(value)))
+    );
 
-    builder.setFilter(likeFilter);
     assertPQLQuery(
         builder.build(),
-        "SELECT span_id FROM SpanEventView "
+        "SELECT duration_millis FROM SpanEventView "
             + "WHERE "
             + viewDefinition.getTenantIdColumn()
             + " = '"
             + TENANT_ID
             + "' "
-            + "AND REGEXP_LIKE(parent_span_id,'')");
```

Comment:
Don't we need a REGEX_LIKE operator as well? Is it blocked at the query/getaway layer as we still calling `handleValueConversionForLiteralExpression`  from the LIKE clause in `convertFilter2String`?

Reply:
I've removed this because `REGEXP_LIKE` doesn't make sense for null or empty string. `col = ''` is the way to check for BYTES columns.